test/unit-tests/marginCalculatorNakedMargin.test.ts

```diff
@@ -0,0 +1,848 @@
+import {
+  MockERC20Instance,
+  CalculatorTesterInstance,
+  MockAddressBookInstance,
+  MockOracleInstance,
+  MockOtokenInstance,
+} from '../../build/types/truffle-types'
+import {
+  createScaledNumber as scaleNum,
+  createScaledBigNumber as scaleBigNum,
+  createVault,
+  createScaledBigNumber,
+  createScaledNumber,
+} from '../utils'
+import {assert} from 'chai'
+import BigNumber from 'bignumber.js'
+
+const {expectRevert, time} = require('@openzeppelin/test-helpers')
+const MockAddressBook = artifacts.require('MockAddressBook.sol')
+const MockOracle = artifacts.require('MockOracle.sol')
+const MockOtoken = artifacts.require('MockOtoken.sol')
+const MockERC20 = artifacts.require('MockERC20.sol')
+const MarginCalculator = artifacts.require('CalculatorTester.sol')
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000'
+
+const expectedRequiredMargin = (
+  shortAmount: number,
+  strikePrice: number,
+  underlyingPrice: number,
+  isPut: boolean,
+  upperBoundValue: number,
+  spotShockValue: number,
+) => {
+  let a, b, marginRequired
+  // if (isPut) {
+  //   a = BigNumber.minimum(strikePrice, spotShockValue.times(underlyingPrice).dividedBy(1e27))
+  //   b = BigNumber.maximum(strikePrice.minus(spotShockValue.times(underlyingPrice).dividedBy(1e27)), new BigNumber(0))
+  //   marginRequired = upperBoundValue.times(a).plus(b).times(shortAmount).dividedBy(1e27)
+  // } else {
+  //   const one = scaleBigNum(1, 27)
+  //   a = BigNumber.minimum(one, strikePrice.dividedBy(underlyingPrice.dividedBy(spotShockValue)))
+  //   b = BigNumber.maximum(one.minus(strikePrice.dividedBy(underlyingPrice.dividedBy(spotShockValue))), new BigNumber(0))
+  //   marginRequired = upperBoundValue.times(a).plus(b).times(shortAmount);
+  // }
```

Comment:
why is this commented out?

Reply:
need to remove it, not using that code, using the one above.